### ISSUE TITLE
Impossible to display more then one Tooltip

### ISSUE THREAD

**Issue by Perlipopette** [StephanWagner/jBox#100]
Hi there !

I'm a total newby in jQuery, so please forgive me :o)

Here's the code someone else put on the site of a client :

	new jBox('Tooltip', {
    attach: '.zlc-infobulle',
    getTitle: 'data-jbox-title',
    content: $('#infobulle-content'),
    closeOnMouseleave: true,
    pointer: 'center:-50',
    maxWidth: 400,
    adjustPosition: true,
    adjustTracker: true
	});

With this code, each time there is two tooltips, the first one is correctly displayed, and also the title of the second, but the content is always the same, so the second tooltip has the content of the first one. 

Could you please help me with this ? What should I put in content: '' so that both of the tooltips display correctly ?

Thanks and have a bright shiny day !

**Comment by StephanWagner** [comment]
Hey Perlipopette,
your tooltip is set up to get the title from the element it is attached to. `getTitle: 'data-jbox-title'` means "look for the attribute `data-jbox-title` in the attached element and set this as the jBox title when the tooltip opens".

The content of the tooltip us defined as `content: $('#infobulle-content')`. This is a jQuery element. So when jBox is created, it will look for an HTML element with the id `infobulle-content` and move this element into the content area of the jBox. It will only do this when it is created. 

If you need different content for each tooltip, you could use the `getContent` option. See this fiddle here: https://jsfiddle.net/StephanWagner/xjLbk1qt/

If you need jBox content with HTML dependent on the attached element, you could create your own content toggler, check out this fiddle: https://jsfiddle.net/StephanWagner/xjLbk1qt/3/

Hope this helped... let me know if you need more assistance.

**Comment by Perlipopette** [comment]
Thank you very much for your quick response and great explanations, everything is clear now ! I have only one question more. Do I have to install an update you posted three days ago to get the HTML content ?

**Comment by Perlipopette** [comment]
Hey again Stephan !

I don't know if your support goes this far, but I would be very gratefull if you had a solution. I chose the HTML version, because there are links ans images to put in those tooltips.

Here's the code the guy put to display the tooltip : 

`if( ! function_exists( 'zlc_single_tarifs_location' ) ) {

  function zlc_single_tarifs_location() {
  
    global $product;
    $product_id = $product->get_id();
    $disponibilite = $product->get_attribute( 'pa_disponibilite' );
    
    // User language
    $langue = get_locale();
    switch( $langue ) {
      case "en_US": 
        $tarifs_title = "Rental rates";
        $caution_title = "Deposit";
        $label_unavailable = "This bicycle is not available for rent";
        $info_txt = "Prices are regressive, the longer you rent, the less expensive it is. For the exact price, enter the number of days and bikes you want below.";
        $see_page = "Please see the page";
        $link_label = "Deposit to rent a bike";
        break;
      case "es_ES": 
        $tarifs_title = "Precios de alquiler";
        $caution_title = "Depósito";
        $label_unavailable = "Esta bicicleta no está disponible al alquiler";
        $info_txt = "Los precios son regresivos, cuanto más tiempo se alquila, menos costoso es. Para conocer el precio exacto, indica la cantidad de días y bicicletas que deseas a continuación.";
        $see_page = "Por favor, consulta la página";
        $link_label = "Depósito para alquilar";
        break;
      default: 
        $tarifs_title = "Prix à la location";
        $caution_title = "Caution";
        $label_unavailable = "Ce véhicule n'est pas disponible à la location";
        $see_page = "Veuillez consulter la page";
        $link_label = "Caution pour la location";
        $info_txt = "Les prix sont régressifs. Plus vous louez longtemps, plus c'est économique. Pour connaître le prix exact, indiquez le nombre de jours et triporteurs souhaités ci-dessous.";
    }
    
    echo '<h3>' . $tarifs_title . '</h3>';
    
    if( empty( $disponibilite ) || ( strpos( $disponibilite, 'En location' ) === false ) ) {
      echo "<p>" . $label_unavailable . "</p>";
    }
    else {
      echo '<h5 class="text-center btn-container text-muted">' . $info_txt . '</h5>';
      
      $jour       = get_post_meta( $product_id, '_prix_par_jour', true );
      $semaine    = get_post_meta( $product_id, '_prix_par_semaine', true );
      $mois       = get_post_meta( $product_id, '_prix_par_mois', true );
      $caution    = get_post_meta( $product_id, '_location_caution', true );
      
      zlc_calculateur_prix_location();
      
      if( $caution ) {
        echo '<table>';
        echo '<tbody>';
        echo '<tr>';
        **echo '<th>'. $caution_title . ' <span class="zlc-infobulle" data-jbox-title="Caution"></span><div id="infobulle-content" style="display:none">' . $see_page . ' : <a href="https://www.tripup.fr/ufaqs/caution-a-la-location/" target="_blank">' . $link_label . '</a></div></div></th>';
        echo '<td>' . $caution . ' &euro;</td>';**
        echo '</tr>';
        echo '</tbody>';
        echo '</table>';
        echo '<p style="font-size: 11px; text-align: center;"><a href="https://www.tripup.fr/download/2019/" target="_blank">Conditions Générales de Location</a></p>';
      }
      
    }
  }

}
`

I only change this line :

`echo '<th>'. $caution_title . ' <span class="zlc-infobulle" data-jbox-title="Caution"></span><div id="infobulle-content" style="display:none">' . $see_page . ' : <a href="https://www.tripup.fr/ufaqs/caution-a-la-location/" target="_blank">' . $link_label . '</a></div></div></th>';`

Into this :

`echo '<th><span class="zlc-infobulle" data-jbox-title="Caution" data-jbox-content="ID1">' . $caution_title . '</span><div class="jbox-content jbox-content-ID1">' . $see_page . ' : <a href="https://www.tripup.fr/ufaqs/caution-a-la-location/" target="_blank">' . $link_label . '</a></div></th>';`

But here's what I get :

![image](https://user-images.githubusercontent.com/36418283/36146729-da4bf0a2-10b5-11e8-9aca-198dbf0a1014.png)

I put the css into jBox.css and into style.ss of the theme, but it doesn't seem to work. 

Could you please help ? I can't contact the guy that did this, so I really hope you can help me.

Have a nice & bright day in Germany !

**Comment by Perlipopette** [comment]
Hey again Stephan !

Where should I place the css ? I tried in style.css of the theme where it doesn't work, and in jBox.css, where it works, but it doesn't display the whole tooltip. I don't know if it's you or the other person that coded this site that did this, but there are other jBox-content in the css, like :

`.jBox-title,
.jBox-content,
.jBox-container {
  position: relative;
  word-break: break-word;
  box-sizing: border-box;
}
`
or

`.jBox-content {
  padding: 8px 10px;
  overflow-x: hidden;
  overflow-y: auto;
  transition: opacity .2s;
}`

Maybe it's the source of the problem ? 

Could you please help ?

**Comment by StephanWagner** [comment]
Hey there, it doesn't really matter where you place it, as long as it is loaded and overwrites any old CSS. If you are worried about the HTML elements as they are displayed by default, you could also just set the CSS at the element. See this updated fiddle: https://jsfiddle.net/StephanWagner/xjLbk1qt/5/

I just moved the `display: none` to the content element, so it is hidden by default, and will only show when jBox opens: `<div class="jbox-content jbox-content-ID1" style="display: none">`

**Comment by Perlipopette** [comment]
PS.
I also tried to change .jBox-content into jBox-contenu, but it's still the same, the whole tooltip is not displaying.

**Comment by StephanWagner** [comment]
Just follow the guide of this fiddle:
https://jsfiddle.net/StephanWagner/xjLbk1qt/5/

The element that opens the tooltip needs to define which content ID should be used:
`<div ... data-jbox-content="ID1">My Tooltip 1</div>`

The content elements need a shared class and a unique class with the ID mentioned above:
`<div class="jbox-content jbox-content-ID1" style="display: none">Content 1</div>`
`<div class="jbox-content jbox-content-ID2" style="display: none">Content 2</div>`

Now you only need to tell your jBox instance to include the content with the shared class `jbox-content`. You can use any classname you like:
```
...
content: $('.jbox-content')
...
```

Then you adjust the onOpen method of your jBox to hide all content elements, and only show the element whith the ID defined in the attached element:
```
  onOpen: function () {
    // Here we are getting the ID of the content to display
    var id = this.source.attr('data-jbox-content');

    // Now we are hiding all content elements
    this.content.find('.jbox-content').css({display: 'none'});
      
    // Then we show the content with the id we just received
    this.content.find('.jbox-content-' + id).css({display: 'block'});
  }
```

**Comment by StephanWagner** [comment]
Then your HTML is not included, first make sure your JavaScript of your jBox looks similar to this:

```
new jBox('Tooltip', {
    attach: '.zlc-infobulle',
    getTitle: 'data-jbox-title',
    // First we move all elements with the
    // class .jbox-content to the content area
    content: $('.jbox-content'), // <-- Make sure here you use the class you've set to your HTML elements, in your example '.jbox-content'
    position: {
    	x: 'right',
    	y: 'center'
    },
    outside: 'x',
    closeOnMouseleave: true,
    //pointer: 'center:-50',
    maxWidth: 400,
    minWidth: 200,
    adjustPosition: true,
    adjustTracker: true,
    
    // When jBox opens we adjust the content accordingly
    onOpen: function () {
    	// Here we are getting the ID of the content to display
    	var id = this.source.attr('data-jbox-content');
      
      // Now we are hiding all contents
			this.content.find('.jbox-content').css({display: 'none'});
      
      // Then we show the content with the id we just received
			this.content.find('.jbox-content-' + id).css({display: 'block'});
    }
```

**Comment by Perlipopette** [comment]
Yes it's the same, I only put your suggestion of jbox-content-html:

`new jBox('Tooltip', {
    attach: '.zlc-infobulle',
    getTitle: 'data-jbox-title',
    // First we move all elements with the
    // class .jbox-content to the content area
    content: $('.jbox-content-html'),
    position: {
    	x: 'right',
    	y: 'center'
    },
    outside: 'x',
    closeOnMouseleave: true,
    //pointer: 'center:-50',
    maxWidth: 400,
    minWidth: 200,
    adjustPosition: true,
    adjustTracker: true,
    
    // When jBox opens we adjust the content accordingly
    onOpen: function () {
    	// Here we are getting the ID of the content to display
    	var id = this.source.attr('data-jbox-content-html');
      
      // Now we are hiding all contents
			this.content.find('.jbox-content-html').css({display: 'none'});
      
      // Then we show the content with the id we just received
			this.content.find('.jbox-content-html-' + id).css({display: 'block'});
    }
  });
	
});
`

**Comment by StephanWagner** [comment]
OK, then now make sure your html has the same classes and attributes (with -html added) and it should work:
```
$out = '<h3>' . $title . '<span class="zlc-infobulle" data-jbox-title="Customisation" data-jbox-content-html="ID2" style="font-size: 18px; margin-top: -10px;"></span><div class="jbox-content-html jbox-content-html-ID2" style="display:none;">Il s’agit d’un adhésif monomère laminé qui est donné pour 2 ou 3 ans.</div></h3>';
```

**Comment by Perlipopette** [comment]
It seem ok too to me, do you see any errors ?

`$out = '<h3>' . $title . '<span class="zlc-infobulle" data-jbox-title="Customisation" data-jbox-content-html="ID2" style="font-size: 18px; margin-top: -10px;"></span><div class="jbox-content-html jbox-content-html-ID2" style="display:none;">Il s’agit d’un adhésif monomère laminé qui est donné pour 2 ou 3 ans.</div></h3>';`

**Comment by StephanWagner** [comment]
Looks good to me too

**Comment by Perlipopette** [comment]
Yes but it's still the same, it doesn't display the content of the tooltip:

![image](https://user-images.githubusercontent.com/36418283/36151001-e4ccd9dc-10c5-11e8-9842-d43a4c2b6e55.png)

**Comment by Perlipopette** [comment]
In Firefox and in Chrome it displays like on the image I just send you, but on Opera, it doesn't display the whole tooltip

**Comment by StephanWagner** [comment]
I just created a new fiddle with the code you posted, it should work just fine: https://jsfiddle.net/StephanWagner/xjLbk1qt/6/

**Comment by Perlipopette** [comment]
Rha I'm so sorry to take all this time to you ! But no, it doesn't work, and it wouldn't take in account the languages with this code, this is why I copied to you the function in witch it is coded.

**Comment by Perlipopette** [comment]
I think the only problem is the display : none.

**Comment by StephanWagner** [comment]
The language shouldn't matter, cause thats PHP. Once rendered as HTML the JavaScript should be fine to work with the HTML. You gat an online example of that issue?

**Comment by Perlipopette** [comment]
Yes, here it is, but I can't leave it like that very long :)

[https://www.tripup.fr/produit/pousse-pousse-asie/]

**Comment by Perlipopette** [comment]
(https://www.tripup.fr/produit/pousse-pousse-asie/)

**Comment by StephanWagner** [comment]
Also I noticed that the language menu is below the red navigation bar:

![screen shot 2018-02-13 at 14 22 06](https://user-images.githubusercontent.com/1636873/36151862-494bdaa4-10c9-11e8-9c41-43f69d733d69.png)

You could fix that with the CSS 
```
header {
  position: relative;
  z-index: 12000;
}
```

**Comment by Perlipopette** [comment]
Wow youo really really ROCK Stephan !!! Thanks it's realy great to find someone this helpfull !

And yes, I noticed it too, I just got on this contract and I'm on test so I planned to do it :o)

Thank you so so so much !

Have a shiney day and a bright life !

**Comment by StephanWagner** [comment]
No worries! Anytime... Feel free to contact me if you need some more help: stephanwagner.me@gmail.com

**Comment by Perlipopette** [comment]
Wowy Wow ! Ok I'll send you a message to keep the mail in my contacts :)

**Comment by Perlipopette** [comment]
Hi there,

I'm happy to contact you again :dancing_women:  please take your time to answer, we got one week until it goes online.

Here's the problem : Although the ID's are different each time, the content of the tooltip is displaying twice.

And here's the link for the page, you would have to scroll until "Simulation d'achat" to see it, under "Options", you have the first example just by "Triporteur 510 NEW" :

https://www.tripup.fr/produit/rosalie-100-2/

Could you please help me with this ?

Thanks !

Have a nice & bright day :o)

Marina

**Comment by StephanWagner** [comment]
Hello again,

I can’t really find the tooltip, there is a jQuery bug:

Uncaught TypeError: jQuery(...) is not a function at tripup.js:159 tripup.smoothscroll.js:43 
Uncaught TypeError: jQuery(...) is not a function at tripup.smoothscroll.js:43

Cheers,
Stephan

**Comment by Perlipopette** [comment]
Hi there,

I'm sorry, I know nothing about jQuery, so could you please help ?

Here's the code for tripup.smoothscroll.js, the line 43 is the last one, the cosing tag: 

`jQuery(function( $ ) {

	'use strict';
	
	// Select all links with hashes
	$('a[href*="#"]')
  // Remove links that don't actually link to anything
  .not('[href="#"]')
  .not('[href="#0"]')
  .click(function(event) {
    // On-page links
    if (
      location.pathname.replace(/^\//, '') == this.pathname.replace(/^\//, '') 
      && 
      location.hostname == this.hostname
    ) {
      // Figure out element to scroll to
      var target = $(this.hash);
      target = target.length ? target : $('[name=' + this.hash.slice(1) + ']');
      // Does a scroll target exist?
      if (target.length) {
        // Only prevent default if animation is actually gonna happen
        event.preventDefault();
        $('html, body').animate({
          scrollTop: (target.offset().top - 70)
        }, 1000, function() {
          // Callback after animation
          // Must change focus!
          var $target = $(target);
          $target.focus();
          if ($target.is(":focus")) { // Checking if the target was focused
            return false;
          } else {
            $target.attr('tabindex','-1'); // Adding tabindex for elements not focusable
            $target.focus(); // Set focus again
          };
        });
      }
    }
  });


})(jQuery);`

And here is a screen capture of the first tooltip: 

![image](https://user-images.githubusercontent.com/36418283/38485607-b83d9d30-3bda-11e8-8e1d-12b342277235.png)


Thank you !

**Comment by Perlipopette** [comment]
Hi Stephen,

Just a little up, so you don't forget me, we are at the end of the week, so I would need a solution now. Could you please help ?

Marina

**Comment by Perlipopette** [comment]
Hi Stephen,

And thank you so much ! I found the error, I put twice the same form, witch won't happen in a real context. 

I'm sorry, but I have a new problem again. It's a WordPress, and I programmed the products page to do the shortcode of the form. But when I just put a shortcode in a WordPress page, the tooltips are all positioned behind the form. 

So I thought maybe I could get a z-index to work, and I put it here:

`// When jBox opens we adjust the content accordingly
    onOpen: function () {
    	// Here we are getting the ID of the content to display
    	var id = this.source.attr('data-jbox-content-html');
      
      // Now we are hiding all contents
			this.content.find('.jbox-content-html').css({display: 'none'});
      
      // Then we show the content with the id we just received
			this.content.find('.jbox-content-html-' + id).css({display: 'block', z-index: '16777271'});
    }`

But then it didn't display any of the tooltips neither on the product page. Could you please help me with this ? Hope this one last time !

Thanks and cheers !
Marina

**Comment by StephanWagner** [comment]
Hmm, not sure about this one,

but jBox can also have a z-Index: https://stephanwagner.me/jBox/options#appearance <https://stephanwagner.me/jBox/options#appearance>

new jBox(’Tooltip’, {
  // … other options

  zIndex: 20000

});

Maybe you can try this?

**Comment by Perlipopette** [comment]
Yes I've tried, but it doesn't display any tooltip any more when I put a zIndex. Do you know why ?

I guess it's the theme or the plugin super forms that does it anyway, so I guess you couldn't help, but I just thought I try and see what I get :o) Maybe you have some magical tools I don't :o)

Thanks !

Have a nice & bright day :o)
Marina
